my-cart/src/app/reducers/products.reducer.ts

```diff
@@ -0,0 +1,47 @@
+import { Product } from "../model/Products.model";
+import { ProductState } from "../products/ProductState";
```

Comment:
I think what we want to maintain is a cart's state, not a product's state. Shouldn't we model the types, actions and reducers around a `Cart` or `UserCart` model?